test/Client.ts

```diff
@@ -47,7 +47,7 @@ describe('Client', () =>
 
     it('should connect to the given server if a parameter is specified', (done) =>
     {
-        let dgram_stub = sandbox.stub(dgram.Socket.prototype, 'send').yields(null, 10);
+        let dgram_stub: sinon.SinonStub<any, void> = sandbox.stub(dgram.Socket.prototype, 'send').yields(null, 10);
```

Comment:
Is `any` the only possible type here?

Reply:
For sinon's stubs and spies, it seems the inferred arguments for `dgram.Socket.send` are not the ones of the right overload:

`dgram.Socket.send` has 6 overloads:
```typescript
send(msg: string | Uint8Array | ReadonlyArray<any>, port?: number, address?: string, callback?: (error: Error | null, bytes: number) => void): void;
send(msg: string | Uint8Array | ReadonlyArray<any>, port?: number, callback?: (error: Error | null, bytes: number) => void): void;
send(msg: string | Uint8Array | ReadonlyArray<any>, callback?: (error: Error | null, bytes: number) => void): void;
send(msg: string | Uint8Array, offset: number, length: number, port?: number, address?: string, callback?: (error: Error | null, bytes: number) => void): void;
send(msg: string | Uint8Array, offset: number, length: number, port?: number, callback?: (error: Error | null, bytes: number) => void): void;
send(msg: string | Uint8Array, offset: number, length: number, callback?: (error: Error | null, bytes: number) => void): void;
```

and Typescript is not able to infer the right one, as stated in [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/36436).

A possible workaround, is to cast dgram.Socket.prototype as an object containing only the right overload:
```typescript
let dgram_stub =
    sandbox.stub(<{
        'send': (msg: string | Uint8Array, offset: number, length: number, port?: number, address?: string, callback?: (error: Error | null, bytes: number) => void) => void
    }>dgram.Socket.prototype, 'send').yields(null, 10);
```

But it's not a clean solution at all.

Here in this code I left `any` because it allowed to accept any type of overload for `dgram.Socket.send` when trying to assert a specific overload call.

Reply:
Clever explanation, thank you :pray: 

We may leave `any` then :ok_man: